alica_engine/include/engine/AlicaEngine.h

```diff
@@ -169,9 +138,9 @@ class AlicaEngine
  * IntRobotID).
  */
 template <class Prototype>
-AgentIDConstPtr AlicaEngine::getId(Prototype& idPrototype) const
+AgentIDConstPtr AlicaEngine::getId(Prototype& idPrototype)
```

Comment:
It seems this change has a pretty big effect, now the AlicaEngine object is non-cosnt accessible in a many places, which looks counter-productive.

Reply:
Having regular members in engine class have caused this major change. Engine is now composed of other modules rather than just placeholder for them. We can decide based on what we want engine to be. In the later case though we were having misleading api's like getId which operated on const engine object and internally mutated other module(AgentIdManager)

Reply:
Yea :S